src/common/transformations/src/transformations/common_optimizations/pull_through_reduce.cpp

```diff
@@ -103,10 +103,11 @@ ov::pass::PullUnsqueezeThroughReduce::PullUnsqueezeThroughReduce() {
 
     const auto input = pattern::any_input(pattern::has_static_rank());
     const auto unsqueeze_axes = pattern::wrap_type<opset9::Constant>();
-    const auto unsqueeze = pattern::wrap_type<opset9::Unsqueeze>({input, unsqueeze_axes}, pattern::has_static_rank());
+    const auto unsqueeze = pattern::wrap_type<opset9::Unsqueeze>({input, unsqueeze_axes}, pattern::consumers_count(1));
     const auto reduce_axes = pattern::wrap_type<opset9::Constant>();
     const auto reduce = pattern::wrap_type<op::util::ArithmeticReductionKeepDims, op::util::LogicalReductionKeepDims>(
-        {unsqueeze, reduce_axes});
+        {unsqueeze, reduce_axes},
+        pattern::consumers_count(1));
```

Comment:
Is `consumers_count(1)` really a limitation here?
as I can see we use `replace_node(m.get_match_root(), new_unsqueeze_node);` , it should handle the case when there are multiple consumers of  Reduction op

Reply:
@itikhono A good catch, thank you. Restrictions removed.